pkg/kubelet/remote/remote_image.go

```diff
@@ -101,7 +103,7 @@ func (r *RemoteImageService) ImageStatus(image *runtimeapi.ImageSpec) (*runtimea
 
 // PullImage pulls an image with authentication config.
 func (r *RemoteImageService) PullImage(image *runtimeapi.ImageSpec, auth *runtimeapi.AuthConfig, podSandboxConfig *runtimeapi.PodSandboxConfig) (string, error) {
-	ctx, cancel := getContextWithCancel()
+	ctx, cancel := getContextWithTimeout(r.imagePullProgressDeadline)
```

Comment:
Are their benefits to adding a test which verifies the timeout is actually respected by the image client? Any initial thoughts on how difficult it would be to do so?